### ISSUE TITLE
requirements: tensorflow>=2.4.1 not found and is required by YOLOv5, attempting auto-update...

### ISSUE THREAD

**Issue by aidemHJC** [ultralytics/yolov5#5465]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Dear Sir,
thanks for your code!
I got the best.pt which can run well in detect.py, and the argument is:
parser.add_argument('--weights', nargs='+', type=str, default='runs/train/exp12/weights/best.pt', help='model path(s)')

then got the best.tflite by export.py, also I want to run detect.py with best.tflite,and the argument is:
parser.add_argument('--weights', nargs='+', type=str, default='runs/train/exp12/weights/best.tflite', help='model path(s)')

I use pip install tensorflow-cpu 2.4.1 and tensorflow-gpu 2.4.1, even i changed to tensorflow-gpu 2.4.3

but the log stays at:**requirements: tensorflow>=2.4.1 not found and is required by YOLOv5, attempting auto-update...**
after minutes later, raise ReadTimeoutError(self._pool, None, "Read timed out.")

thanks,BR

### Additional

my conda list shows:
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
absl-py                   0.13.0                   pypi_0    pypi
astunparse                1.6.3                    pypi_0    pypi
ca-certificates           2021.5.30            ha878542_0    conda-forge
cachetools                4.2.2                    pypi_0    pypi
certifi                   2021.5.30                pypi_0    pypi
charset-normalizer        2.0.4                    pypi_0    pypi
coremltools               5.0                      pypi_0    pypi
cycler                    0.10.0                   pypi_0    pypi
cython                    0.29.24                  pypi_0    pypi
dataclasses               0.8                      pypi_0    pypi
flatbuffers               1.12                     pypi_0    pypi
gast                      0.3.3                    pypi_0    pypi
google-auth               1.35.0                   pypi_0    pypi
google-auth-oauthlib      0.4.6                    pypi_0    pypi
google-pasta              0.2.0                    pypi_0    pypi
grpcio                    1.32.0                   pypi_0    pypi
h5py                      2.10.0                   pypi_0    pypi
idna                      3.2                      pypi_0    pypi
importlib-metadata        4.8.1                    pypi_0    pypi
keras-preprocessing       1.1.2                    pypi_0    pypi
kiwisolver                1.3.1                    pypi_0    pypi
ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
libffi                    3.4.2                h9c3ff4c_1    conda-forge
libgcc-ng                 11.1.0               hc902ee8_8    conda-forge
libgomp                   11.1.0               hc902ee8_8    conda-forge
libstdcxx-ng              11.1.0               h56837e0_8    conda-forge
markdown                  3.3.4                    pypi_0    pypi
matplotlib                3.3.4                    pypi_0    pypi
mpmath                    1.2.1                    pypi_0    pypi
ncurses                   6.2                  h58526e2_4    conda-forge
numpy                     1.19.5                   pypi_0    pypi
oauthlib                  3.1.1                    pypi_0    pypi
onnx                      1.10.2                   pypi_0    pypi
opencv-python             4.5.3.56                 pypi_0    pypi
openssl                   1.1.1l               h7f98852_0    conda-forge
opt-einsum                3.3.0                    pypi_0    pypi
packaging                 21.2                     pypi_0    pypi
pandas                    1.1.5                    pypi_0    pypi
pillow                    8.3.2                    pypi_0    pypi
pip                       21.2.4             pyhd8ed1ab_0    conda-forge
protobuf                  3.17.3                   pypi_0    pypi
pyasn1                    0.4.8                    pypi_0    pypi
pyasn1-modules            0.2.8                    pypi_0    pypi
pyparsing                 2.4.7                    pypi_0    pypi
python                    3.6.13          hb7a2778_1_cpython    conda-forge
python-dateutil           2.8.2                    pypi_0    pypi
python_abi                3.6                     2_cp36m    conda-forge
pytz                      2021.1                   pypi_0    pypi
pyyaml                    5.4.1                    pypi_0    pypi
readline                  8.1                  h46c0cb4_0    conda-forge
requests                  2.26.0                   pypi_0    pypi
requests-oauthlib         1.3.0                    pypi_0    pypi
rsa                       4.7.2                    pypi_0    pypi
scikit-learn              0.19.2                   pypi_0    pypi
scipy                     1.5.4                    pypi_0    pypi
seaborn                   0.11.2                   pypi_0    pypi
setuptools                58.0.4           py36h5fab9bb_0    conda-forge
six                       1.15.0                   pypi_0    pypi
sqlite                    3.36.0               h9cd32fc_1    conda-forge
sympy                     1.9                      pypi_0    pypi
tensorboard               2.6.0                    pypi_0    pypi
tensorboard-data-server   0.6.1                    pypi_0    pypi
tensorboard-plugin-wit    1.8.0                    pypi_0    pypi
tensorflow-estimator      2.4.0                    pypi_0    pypi
tensorflow-gpu            2.4.3                    pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
thop                      0.0.31-2005241907          pypi_0    pypi
tk                        8.6.11               h27826a3_1    conda-forge
torch                     1.9.0                    pypi_0    pypi
torchvision               0.10.0                   pypi_0    pypi
tqdm                      4.62.2                   pypi_0    pypi
typing-extensions         3.7.4.3                  pypi_0    pypi
urllib3                   1.26.6                   pypi_0    pypi
werkzeug                  2.0.1                    pypi_0    pypi
wheel                     0.37.0             pyhd8ed1ab_1    conda-forge
wrapt                     1.12.1                   pypi_0    pypi
xz                        5.2.5                h516909a_1    conda-forge
zipp                      3.5.0                    pypi_0    pypi
zlib                      1.2.11            h516909a_1010    conda-forge

requirements.txt:
# pip install -r requirements.txt

# Base ----------------------------------------
matplotlib>=3.2.2
numpy>=1.18.5
opencv-python>=4.1.2
Pillow>=8.0.0
PyYAML>=5.3.1
scipy>=1.4.1
torch>=1.7.0
torchvision>=0.8.1
tqdm>=4.41.0

# Logging -------------------------------------
tensorboard>=2.4.1
# wandb

# Plotting ------------------------------------
seaborn>=0.11.0
pandas

# Export --------------------------------------
# coremltools>=4.1  # CoreML export
# onnx>=1.9.0  # ONNX export
# onnx-simplifier>=0.3.6  # ONNX simplifier
# scikit-learn==0.19.2  # CoreML quantization
tensorflow>=2.4.1  # TFLite export
# tensorflowjs>=3.9.0  # TF.js export

# Extras --------------------------------------
# Cython  # for pycocotools https://github.com/cocodataset/cocoapi/issues/172
# pycocotools>=2.0  # COCO mAP
# albumentations>=1.0.3
thop  # FLOPs computation

**Comment by aidemHJC** [comment]
![image](https://user-images.githubusercontent.com/49936637/140026089-287dd96d-0d21-492b-a269-317514e7bf3b.png)
![image](https://user-images.githubusercontent.com/49936637/140026208-6cce4433-e965-4569-96a5-9dff93814c96.png)
![image](https://user-images.githubusercontent.com/49936637/140026303-4185c55a-4b5f-4c85-8fd2-92776e7f5f00.png)

**Comment by glenn-jocher** [comment]
@aidemHJC good news 😃! Your original issue may now be fixed ✅ in PR #5476. This PR removes TF checks, as different installation methods are hard to all verify as in your example.

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
